bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ChannelItemProvider.java

```diff
@@ -189,10 +189,13 @@ private void createItemForLink(ItemChannelLink link) {
         Channel channel = thingRegistry.getChannel(link.getUID());
         if (channel != null) {
             Item item = null;
-            for (ItemFactory itemFactory : itemFactories) {
-                item = itemFactory.createItem(channel.getAcceptedItemType(), link.getItemName());
-                if (item != null) {
-                    break;
+            // Only create an item for state channels
+            if (channel.getKind() == ChannelKind.STATE) {
```

Comment:
don't we need for backward compatibility `(channel.getKind() == null) || channel.getKind() == ChannelKind.STATE)` here?